smack-core/src/main/java/org/jivesoftware/smack/XMPPConnection.java

```diff
@@ -141,6 +142,16 @@ public interface XMPPConnection {
      */
     EntityFullJid getUser();
 
+    /**
+     * Returns the local address currently in use for this connection, or <code>null</code> if
+     * this is invalid the type of underlying connection.
+     *
+     * @return the local address currently in use for this connection
+     */
+    default InetAddress getLocalAddress() {
```

Comment:
I tend to have this not come with a default implementation, as it makes it obvious to creators of XMPPConnection classes that there is some behavior that, even though not strictly required, they may want to implement.

Reply:
OK, I see the benefit of that but the result of this is the default has been copied and pasted into `ModularXmppClientToServerConnection`, `XMPPBOSHConnection` and `DummyConnection`.